From ce6e9d4d5bcadfb4f65f5bd4e13a1068ed611dfe Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Mon, 2 Aug 2021 13:15:29 +0000
Subject: [PATCH] version.sh: output short mc git version to prevent full rebuild

When using exact git version in AC_INIT, every git commit result in different AC_PACKAGE_VERSION and AC_PACKAGE_STRING macros at config.h
hence this will cause a full rebuild see #2252

4.8.26-217-ge819ed742 -> 4.8.26-git
4.8.27-pre1-1-gcbc214da8 -> 4.8.27-pre1-git

TODO tests needed for final version release workflow!

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 version.sh | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/version.sh b/version.sh
index 5d7aa1615..0565b7448 100755
--- a/version.sh
+++ b/version.sh
@@ -42,7 +42,7 @@ mc_print_version(){
 #endif
 EOF
     fi
-    echo "${CURR_MC_VERSION}"
+    echo "${SHOR_MC_VERSION}"
     exit
 }
 
@@ -59,11 +59,13 @@ src_top_dir="$1"
 VERSION_FILE="${src_top_dir}/mc-version.h"
 PREV_MC_VERSION="unknown"
 CURR_MC_VERSION="${PREV_MC_VERSION}"
+SHOR_MC_VERSION="${PREV_MC_VERSION}"
 
 if [ -r "${VERSION_FILE}" ]
   then
     PREV_MC_VERSION=`sed -n 's/^#define MC_CURRENT_VERSION "\(.*\)"$/\1/p' "${VERSION_FILE}"`
     CURR_MC_VERSION="${PREV_MC_VERSION}"
+    SHOR_MC_VERSION="${PREV_MC_VERSION}"
 fi
 
 git_head=`git --git-dir "${src_top_dir}/.git" rev-parse --verify HEAD 2>/dev/null`
@@ -71,10 +73,16 @@ git_head=`git --git-dir "${src_top_dir}/.git" rev-parse --verify HEAD 2>/dev/nul
 
 # try to store sha1
 CURR_MC_VERSION="${git_head}"
+SHOR_MC_VERSION="${CURR_MC_VERSION}"
 
 new_version=`git --git-dir "${src_top_dir}/.git" describe --always 2>/dev/null`
 [ -z "${new_version}" ] && mc_print_version
 
 # store pretty tagged version
 CURR_MC_VERSION="${new_version}"
+SHOR_MC_VERSION="${CURR_MC_VERSION}"
+
+# stop full rebuild by using not-exact git version string in config.h, see #2252, #4266
+SHOR_MC_VERSION=`git --git-dir "${src_top_dir}/.git" describe --always --abbrev=0 2>/dev/null`-git
+
 mc_print_version
